tests/test_workers/test_tasks/test_build.py

```diff
@@ -438,6 +439,8 @@ def test_prepare_request_for_build(
         mock_gri.side_effect = [binary_image_resolved, from_index_resolved]
         mock_gia.side_effect = [expected_arches, from_index_arches]
         expected_payload_keys.add('from_index_resolved')
+        mock_gil.return_value = 'v4.6'
```

Comment:
This is problematic since `mock_gil.side_effect` is set below. You'll likely need to create a list of return values and then set `mock_gil.side_effect` to it before `_prepare_request_for_build` is called.

Reply:
For my own edification can you explain why it is a problem and the tests are passing as I am not following why it would be a problem?

Reply:
@shawn-hurley this only works because the one test that sets `bundles` in the `pytest.parametrize` call does not set `from_index`. If a test is later added that sets `from_index` and `bundles`, it will fail because the `side_effect` only has enough values for the `get_image_label` calls for each bundle.

Reply:
I added the test that you described and I updated the test to handle it.